### PR TITLE
!htt #19674 Audit HTTP headers and mark correctly as response or request

### PR DESCRIPTION
**Pull request by johanandren**
Refs #19674 

Went through each http header and read the RFC entry on each to figure out if they were correctly marked as request, response or both. I feel that it is possible that some were incorrectly marked as both on purpose, so some eyes from @jrudolph or @sirthias would be great.

### CODE REVIEW

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala [hunk 1]
```diff
@@ -559,7 +559,7 @@ object Link extends ModeledCompanion[Link] {
   def apply(values: LinkValue*): Link = apply(immutable.Seq(values: _*))
   implicit val valuesRenderer = Renderer.defaultSeqRenderer[LinkValue] // cache
 }
-final case class Link(values: immutable.Seq[LinkValue]) extends jm.headers.Link with RequestResponseHeader {
+final case class Link(values: immutable.Seq[LinkValue]) extends jm.headers.Link with ResponseHeader {
```
Comment (johanandren) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala]: Especially this one I found hard to see from the RFC if it actually is ok as request header, but couldn't find any references online to anyone using it in requests so I changed it.

**Comment by johanandren** [comment]
Thanks @sirthias exactly the kind of feedback I was looking for. Will update according to your comments.

**Comment by drewhk** [comment]
LGTM

**Comment by viktorklang** [comment]
Perhaps add test which verify this?

File: akka-http-core/src/test/scala/akka/http/scaladsl/model/headers/HeaderSpec.scala [hunk 2]
```diff
@@ -5,15 +5,15 @@
 package akka.http.scaladsl.model.headers
 
 import akka.http.impl.util._
-import org.scalatest.{ FreeSpec, MustMatchers }
+import org.scalatest._
 
 import akka.http.scaladsl.model._
 
-class HeaderSpec extends FreeSpec with MustMatchers {
+class HeaderSpec extends FreeSpec with Matchers {
```
Comment (johanandren) [inline akka-http-core/src/test/scala/akka/http/scaladsl/model/headers/HeaderSpec.scala]: had to change to get access to the `shouldBe` matcher without too much fuss.

**Comment by 2m** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM, thanks @johanandren !

**Pull request merged**